pkg/imgutil/pull/pull.go

```diff
@@ -30,6 +30,10 @@ import (
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+type RemoteSnapshotterFlags struct {
+	SociIndexDigest string
+}
```

Comment:
can we move this to `pkg/api/types`? I think this is not only related to `pull` (i.e., we can also use these as part of `run` command right`)?

Reply:
Thanks. This makes a lot more sense and adds a couple of imports but makes the code much easier to follow.

> (i.e., we can also use these as part of run command right)

For this case specifically, I don't think so. Technically, since nerdctl attempts to pull images it doesn't have locally, it could be passed in for that purpose, and at that point is there a reason to not just pull first?

But other remote snapshotters can certainly leverage this in the future, so I think it's still a good place for it to reside.

Reply:
I could move it to types, but I still had to keep this in. By importing the types package everywhere I needed it, I ended up creating a circular dependency when importing it into `imgutil.go` due to `network_types.go` in types importing the `netutil` package, which somehow created a circular dependency.

Reply:
If there is circular dependency, it's okay to put `RemoteSnapshotterFlags` in `imgutil` pkg directly. And in `pkg/api/types` you can just use as `imgutil.RemoteSnapshotterFlags`.

Reply:
The issue is that pull.go cannot import imgutil without creating another circular dependency. Perhaps I can have it live in snapshotter.go somehow? I'm not sure exactly how that might look, but if I do that it might solve this issue since it shares the same package. What do you think?